ambry-cloud/src/main/java/com.github.ambry.cloud/VcrRequests.java

```diff
@@ -74,14 +78,22 @@ protected ServerErrorCode validateRequest(PartitionId partition, RequestOrRespon
       metrics.badRequestError.inc();
       return ServerErrorCode.Bad_Request;
     }
-
-    // 2. check if partition is handled by this vcr node
-    return skipPartitionAvailableCheck ? ServerErrorCode.No_Error
-        : storeManager.checkLocalPartitionStatus(partition, null);
+    return ServerErrorCode.No_Error;
   }
 
   @Override
   protected long getRemoteReplicaLag(Store store, long totalBytesRead) {
     return -1;
   }
+
+  @Override
+  protected Map<PartitionId, ReplicaId> createLocalPartitionToReplicaMap() {
+    //Vcr should be able to handle requests for all partitions
+    List<? extends PartitionId> partitionIds = clusterMap.getAllPartitionIds(null);
+    Map<PartitionId, ReplicaId> partitionToReplica = new HashMap<>();
+    for (PartitionId partitionId : partitionIds) {
+      partitionToReplica.put(partitionId, new CloudReplica(partitionId, currentNode));
+    }
+    return partitionToReplica;
+  }
```

Comment:
Can simplify this to:
`return partitionIds.stream().collect(Collectors.toMap(Function.identity(),
    partitionId -> new CloudReplica(partitionId, currentNode)));`

Reply:
fixed.